### PR TITLE
Permit alternate uid/gid/mode for CVMFS cache

### PR DESCRIPTION
**Pull request by jcpunk**
#### Pull Request (PR) description
This permits using a non-hardcoded owner/group/mode for the cvmfs cache

### CODE REVIEW

**Comment by jcpunk** [comment]
Oddly I've noticed that the caches made under `/var/lib/cvmfs` all seem to have a group id of `root` and an owner of `cvmfs`.  I'd have expected the group to also be `cvmfs`?

**Comment by traylenator** [comment]
Hi, Are you able to update the REFERENCE ?

File: manifests/install.pp [hunk 1]
```diff
@@ -33,9 +33,9 @@
   if $cvmfs_cache_base != $default_cvmfs_cache_base {
     file { $cvmfs_cache_base:
       ensure  => directory,
-      owner   => cvmfs,
-      group   => cvmfs,
-      mode    => '0700',
+      owner   => $cvmfs_cache_owner,
```
Comment (traylenator) [inline manifests/install.pp:36]: Would be good to have some tests.

Comment (jcpunk) [reply manifests/install.pp]: I've got to be doing something obvious wrong in the unit test I just wrote, but I don't see it...

**Comment by traylenator** [comment]
> Oddly I've noticed that the caches made under `/var/lib/cvmfs` all seem to have a group id of `root` and an owner of `cvmfs`. I'd have expected the group to also be `cvmfs`?

Yes it seems so , that as been wrong since forever:

```
# rpm -qp --dump cvmfs-2.10.1-1.el9.x86_64.rpm  | grep '/var/lib/cvmfs'
/var/lib/cvmfs 0 1676972196 0000000000000000000000000000000000000000000000000000000000000000 040700 cvmfs cvmfs 0 0 0 X
```
good catch.

**Comment by traylenator** [comment]
Don't you also need to or at least allow to change the user cvmfs as running as :

* **CVMFS_USER**  - The user name that owns and manipulates the files inside the repository.

From https://cvmfs.readthedocs.io/en/stable/apx-parameters.html ?

Not sure we expose that it in configuration.

Would good to run an acceptance test of this but it may actually be tricky since a switch of user on a running system is maybe hard. Will be easier after #168 since that contains an explicit `cvmfs_config kiallall` at the start of each test.

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -497,6 +497,17 @@
             end
           end
 
+          context 'cvmfs_cache_base set to non-defaults' do
+            let(:params) do
```
Comment (traylenator) [inline spec/classes/init_spec.rb:501]: You can use the `super()` trick used in the same file to extend the minimal parameters.

**Comment by jcpunk** [comment]
> > Oddly I've noticed that the caches made under `/var/lib/cvmfs` all seem to have a group id of `root` and an owner of `cvmfs`. I'd have expected the group to also be `cvmfs`?
> 
> Yes it seems so , that as been wrong since forever:
> 
> ```
> # rpm -qp --dump cvmfs-2.10.1-1.el9.x86_64.rpm  | grep '/var/lib/cvmfs'
> /var/lib/cvmfs 0 1676972196 0000000000000000000000000000000000000000000000000000000000000000 040700 cvmfs cvmfs 0 0 0 X
> ```
> 
> good catch.

at least I've not lost my mind :) what should happen next on that?

**Comment by traylenator** [comment]
> > > Oddly I've noticed that the caches made under `/var/lib/cvmfs` all seem to have a group id of `root` and an owner of `cvmfs`. I'd have expected the group to also be `cvmfs`?
> > 
> > 
> > Yes it seems so , that as been wrong since forever:
> > ```
> > # rpm -qp --dump cvmfs-2.10.1-1.el9.x86_64.rpm  | grep '/var/lib/cvmfs'
> > /var/lib/cvmfs 0 1676972196 0000000000000000000000000000000000000000000000000000000000000000 040700 cvmfs cvmfs 0 0 0 X
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > good catch.
> 
> at least I've not lost my mind :) what should happen next on that?

To be safe we do in two stages:
* Add this patch preserving defaults.
* Add a new patch changing the defaults.

Will make testing the 2nd one is real life very easy.

**Review by bastelfreak — APPROVED** [review]

File: manifests/init.pp [hunk 3]
```diff
@@ -114,6 +117,9 @@
   Variant[Enum['auto'],Integer] $cvmfs_quota_limit                    = 1000,
   Float   $cvmfs_quota_ratio                                          = 0.85,
   Stdlib::Absolutepath $cvmfs_cache_base                              = '/var/lib/cvmfs',
+  String[1] $cvmfs_cache_owner                                        = 'cvmfs',
+  String[1] $cvmfs_cache_group                                        = 'cvmfs',
+  String[1] $cvmfs_cache_mode                                         = '0700',
```
Comment (traylenator) [inline manifests/init.pp:122]:
```suggestion
  Stdlib::Filemode $cvmfs_cache_mode                                         = '0700',
```

Will need a reference regenerate am afraid.

File: REFERENCE.md [hunk 4]
```diff
@@ -192,6 +195,30 @@ Location of the CVMFS cache base
...
+
+Data type: `String[1]`
+
+expected group of cvmfs cache
+
+Default value: `'cvmfs'`
+
+##### <a name="-cvmfs--cvmfs_cache_mode"></a>`cvmfs_cache_mode`
+
+Data type: `String[1]`
```
Comment (traylenator) [inline REFERENCE.md:216]:
```suggestion
Data type: `Stdlib::Filemode`
```

**Review by traylenator — CHANGES_REQUESTED** [review]
Just change that one type for the mode and good to go.

File: manifests/init.pp [hunk 5]
```diff
@@ -114,6 +117,9 @@
   Variant[Enum['auto'],Integer] $cvmfs_quota_limit                    = 1000,
   Float   $cvmfs_quota_ratio                                          = 0.85,
   Stdlib::Absolutepath $cvmfs_cache_base                              = '/var/lib/cvmfs',
+  String[1] $cvmfs_cache_owner                                        = 'cvmfs',
+  String[1] $cvmfs_cache_group                                        = 'cvmfs',
+  Stdlib::Filemode $cvmfs_cache_mode                                         = '0700',
```
Comment (traylenator) [inline manifests/init.pp:122]:
```suggestion
  Stdlib::Filemode $cvmfs_cache_mode                                  = '0700',
```

**Review by traylenator — APPROVED** [review]

**Comment by jcpunk** [comment]
> Will make testing the 2nd one is real life very easy.

With 10.0.0 live, I just didn't want this to get lost in the shuffle.

**Pull request merged**